blog/2021-06-14-stryker4s-sbt-performance.mdx

```diff
@@ -0,0 +1,151 @@
+
+- ✨ New testrunner boasting improved performance
+- 🔀 Concurrency for even faster performance
+- 🕵️‍♀️ Coverage analysis to skip mutants without coverage
+- ⏱️ Timeout detection in tests
+- 👽 New mutator: regular expressions. Thanks to [Weapon regeX](https://github.com/stryker-mutator/weapon-regex/)!
+
+In the rest of this blog post, we'll do a deep-dive into the increased performance. Let's start by taking a look at the older version of Stryker4s.
+
+## 🐌 The old
+
+In the older sbt plugin, there was a lot of overhead in running your tests. As a quick reminder, Stryker4s uses [mutation switching](https://stryker-mutator.io/blog/mutation-switching/), so it only has to compile your code once, instead of for each mutant. After mutating your source code, Stryker4s would essentially just call `sbt test` with the mutated code for every mutant. This means sbt would:
+
+1. look for code changes (which there never are).
+1. make sure all compiled code is up-to-date (which it always is).
+1. Start a new Java process with your class files, dependencies, and a small sbt entry point library.
+1. Run your tests and report the results back to the main sbt process.
+1. Repeat for the next mutant.
+
+import Mermaid from '@theme/Mermaid';
+
+<Mermaid
+  _key="sbt-old"
+  chart={`
+sequenceDiagram
+  participant sbt
+  participant Stryker4s
+  participant sbt test process
+  Stryker4s->>sbt: Compile mutated code
+  loop For each mutant
+    Stryker4s->>sbt: executeTests
+    sbt->>sbt: Check for changes?
+    sbt->>+sbt test process: Start new process and run tests
+    sbt test process->>-sbt: Report test results
+    sbt->>Stryker4s: Tests successful Y/N
+  end
+`}
+/>
+
+Doing those checks and starting a new process make a lot of sense for a development workflow, but it's a lot of overhead for mutation testing. This came from the original implementation of the sbt plugin and hasn't changed much in the two years that the plugin has existed. Time for some improvements!
+
+## 🐣 The new
+
+It's clear the process can be optimized. Once Stryker4s is set up and all mutations have been compiled, the only thing we really want to do is run some tests and get a result. Sbt doesn't give plugins full control over the entire test process, but it does have all the information about _what_ your project looks like. This is something we can work with!
+
+### 🏗️ Meet: build info
+
+The first thing we need to do is figure out what tests your project contains. One of the great things about sbt is its standardization and extensibility as a build tool. Because sbt knows exactly what your project looks like, plugins can retrieve all this information. To find out what tests you have, Stryker4s uses three pieces of information from sbt:
```

Comment:
Maybe instead of "we" and "your" we can use "Stryker4s" and "our"?

I.e.: The first thing Stryker4s needs to do is figure out what tests our project contains.